File: Second Semester(Java and C++)/List 3/C++/Rhombus.cpp
#include "Rhombus.hpp"

#include <stdexcept>

Rhombus::Rhombus(int side, int angle) noexcept(false) {
    if (angle < 1 || angle > 179) {
        throw std::invalid_argument("Rhombus' angle must be in range <1, 179>");
    }

    for (int i = 0; i < 4; i++) sides.push_back(side);
    this->angle = angle;
}

std::string Rhombus::getName() noexcept(true) {
    return "Rhombus";
}

File: Second Semester(Java and C++)/List 3/Java/Task 1/Rhombus.java
public class Rhombus extends Quadrangle {
    public Rhombus(int side, int angle) throws IllegalArgumentException {
        if (angle < 1 || angle > 179) {
            throw new IllegalArgumentException("This is not a rhombus!");
        }

        for (int i = 0; i < 4; i++) sides.add(side);
        this.angle = angle;
    }

    @Override
    public String getName() {
        return "Rhombus";
    }
}
